04/app.js

```diff
@@ -4,52 +4,78 @@ import { createRoot } from 'react-dom/client';
 const root = createRoot(document.querySelector('#root'));
 
 class App extends React.Component {
-    state = { 
+    state = {
         firstName: '',
         lastName: '',
         searchQuery: '',
         users: ['Jan Kowalski', 'Michał Nowak'],
     }
 
     renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
+        const { users, searchQuery } = this.state;
+        let usersArray = users;
+
+        if (searchQuery.length > 0) {
+            usersArray = this.filterUsers();
+        }
+
+        return usersArray.map(user => {
             return (
-                <li onClick={ this.clickHandler }>
-                    { name }
+                <li
+                    key={user}
+                    onClick={this.clickHandler}>
+                    {user}
                 </li>
             );
         });
     }
 
+    filterUsers() {
+        const { users, searchQuery } = this.state;
+
+        return users.filter(user => {
+            const userLowerCase = user.toLowerCase();
+            const searchQueryLowerCase = searchQuery.toLowerCase();
+
+            if (userLowerCase.includes(searchQueryLowerCase)) return user
```

Comment:
Pamiętaj, że tutaj zwracamy `true/false` w `.filter()` więc prościej będzie po prostu: `return userLowerCase.includes(searchQueryLowerCase)` dodatkowo unikamy błędu gdy `user = null`, `false`, `''` itp. - bo to wartości "fałszywe" (ang. falsy)